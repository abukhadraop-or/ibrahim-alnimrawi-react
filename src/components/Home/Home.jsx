import Articles from 'components/Articles/Articles';
import Banner from 'components/Banner/Banner';
import {
  Container,
  FeedLi,
  FeedUl,
  MediaContainer,
  SubContainer,
} from 'components/Home/home.style';
import Tags from 'components/Tags/Tags';
import React, { useState } from 'react';

/**
 * Render a <Home> component
 *
 * @return {JSX.Element}
 */
const Home = () => {
  const [tag, setTag] = useState('');
  const [clicked, setClicked] = useState(false);

  /**
   * Set the new tag name and check if the new tag is the same.
   *
   * @param {string} tagName The name of selected tag.
   */
  const onTagChange = (tagName) => {
    if (tag === tagName) {
      setTag('');
      setClicked(true);
    } else {
      setTag(tagName);
      setClicked(false);
    }
  };

  return (
    <Container>
      <Banner />
      <MediaContainer>
        <SubContainer>
          <FeedUl>
            <FeedLi>Global Feed</FeedLi>
          </FeedUl>
          <Articles tag={tag} />
        </SubContainer>
        <Tags
          onTagChange={(tagName) => onTagChange(tagName)}
          clicked={clicked}
          selectedTag={tag}
        />
      </MediaContainer>
    </Container>
  );
};

export default Home;
