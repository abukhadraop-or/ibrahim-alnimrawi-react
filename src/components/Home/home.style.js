import colors from 'style/colors';
import mediaQueries from 'style/media-queries';
import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
`;

export const FeedLi = styled.li`
  border-bottom: 0.188rem solid ${colors.moderateLimeGreen};
  box-sizing: inherit;
  color: ${colors.moderateLimeGreen};
  cursor: default;
  margin-bottom: 0;
  margin-right: 0.5rem;
`;

export const FeedUl = styled.ul`
  display: flex;
  list-style: none;
  margin-bottom: 0;
  margin-left: 2.5rem;
`;

export const MediaContainer = styled.div`
  display: flex;
  justify-content: space-between;

  @media (max-width: ${mediaQueries.mobile}) {
    flex-direction: column-reverse;
  }
`;

export const SubContainer = styled.div`
  width: 100%;
`;
