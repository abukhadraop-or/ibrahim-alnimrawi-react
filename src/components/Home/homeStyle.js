import styled from "styled-components";
export const Container = styled.div`
  display: grid;
`;
export const FeedLi = styled.li`
  border-bottom: 0.188rem solid #5cb85c;
  box-sizing: inherit;
  color: #5cb85c;
  cursor: default;
  margin-bottom: 0;
  margin-right: 0.5rem;
`;
export const FeedUl = styled.ul`
  display: flex;
  list-style: none;
  margin-bottom: 0;
  margin-left: 2.5rem;
`;
