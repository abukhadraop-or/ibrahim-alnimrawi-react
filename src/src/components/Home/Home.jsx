import Articles from "components/Articles/Articles";
import Banner from "components/Banner/Banner";
import {
  Container,
  FeedLi,
  FeedUl,
  MediaContainer,
} from "components/Home/home.style";
import Tags from "components/Tags/Tags";
import React, { useState } from "react";
import { ArticleSection } from "../Articles/articles.style";

/**
 * Render a <Home> component
 *
 * @return {JSX.Element}
 */
const Home = () => {
  const [tag, setTag] = useState("");
  const [unClick, setUnClick] = useState(false);

  /**
   *Set the new tag name and check if the new tag is the same.
   *
   * @param {string} tagName The name of selected tag.
   */
  const handleTag = (tagName) => {
    if (tag === tagName) {
      setTag("");
      setUnClick(true);
    } else {
      setTag(tagName);
      setUnClick(false);
    }
  };

  return (
    <Container>
      <Banner />
      <MediaContainer>
        <div>
          <FeedUl>
            <FeedLi>Global Feed</FeedLi>
          </FeedUl>
          <ArticleSection>
            <Articles tag={tag} />
          </ArticleSection>
        </div>
        <Tags onTagChange={(tagName) => handleTag(tagName)} unClick={unClick} />
      </MediaContainer>
    </Container>
  );
};

export default Home;
