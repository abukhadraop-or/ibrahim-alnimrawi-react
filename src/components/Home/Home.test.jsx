import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

Enzyme.configure({ adapter: new Adapter() });

describe('<Home />', () => {
  it('should be defined', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('should be change', () => {
  //   jest.mock('components/Articles/Articles', () => () => {
  //     <div />;
  //   });
  //   jest.mock('components/Tags/Tags', () => () => {
  //     returned();
  //   });
  // const wrapper = mount(<Home />);
  // });
});
