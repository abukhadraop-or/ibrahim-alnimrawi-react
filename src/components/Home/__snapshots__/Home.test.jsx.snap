// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match the snapshot 1`] = `
.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  color: #373a3c;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 3rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  width: 8rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c10 {
  background-color: #f3f3f3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  margin-right: 3rem;
  max-width: 20rem;
  padding-bottom: 2rem;
}

.c9 {
  border: none;
  color: #2F3236;
  margin: 1rem 5rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #5cb85c;
  border: none;
  box-shadow: inset 0 0.5rem 0.5rem -0.5rem rgb(0 0 0 / 30%), inset 0 -0.5rem 0.5rem -0.5rem rgb(0 0 0 / 30%);
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 2rem;
  padding-top: 0;
  padding: 1rem;
}

.c2 {
  font-size: 3.5rem;
  font-weight: 700;
  margin-bottom: 0.5rem;
  text-shadow: 0 0.063rem 0.188rem rgb(0 0 0 / 30%);
}

.c3 {
  font-size: 1.5rem;
  font-weight: 300;
  margin-bottom: 0;
  text-align: center;
}

.c0 {
  display: grid;
}

.c7 {
  border-bottom: 0.188rem solid #5cb85c;
  box-sizing: inherit;
  color: #5cb85c;
  cursor: default;
  margin-bottom: 0;
  margin-right: 0.5rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin-bottom: 0;
  margin-left: 2.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  width: 100%;
}

@media (max-width:34.375rem) {
  .c10 {
    margin: 0 auto;
    width: 100%;
  }
}

@media (max-width:34.375rem) {
  .c4 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      conduit
    </h1>
    <p
      className="c3"
    >
      A place to share your knowledge.
    </p>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <ul
        className="c6"
      >
        <li
          className="c7"
        >
          Global Feed
        </li>
      </ul>
      <div
        className="c8"
      >
        <div
          className="c9"
        >
          No articles are here... yet.
        </div>
      </div>
    </div>
    <div
      className="c10"
    >
      <div
        className="c11"
      >
        Popular Tags
      </div>
      <div
        className="c12"
      />
    </div>
  </div>
</div>
`;
